### DISCUSSION TITLE
Live tracking with turn by turn navigation.

### DISCUSSION THREAD

**Discussion by jesussmile** [liodali/osm_flutter#96] [Q&A]
So far I am able to implement more or less features you have mentioned. Thank you for that, Is it possible to implement features like live tracking and turn by turn navigation with information of distance and time remaining etc ?  If so, could you please tell me where I could look into or how I could implement it ?

**Comment by liodali** [comment]
it's difficult with osmdroid but i'm already looking to see how i can add that feature
for now you can just keep road visible and zoom in to user location
and also i will add how to get instrection of route like google maps and other provider.

**Comment by liodali** [comment]
i think for such type of feature we need to go with our infrastructure to get accurate resultat
we can combine osmnavigation with valhalla project(https://github.com/valhalla/) to improve tracking of the user route 
and match it with original route or recalculate the route.
it need lot of research to implement this feature

**Reply by jesussmile** [reply]
Here is a free service that a lot of osm developers use for routing [https://routing.openstreetmap.de/about.html](url) sure! You must have known about it already.

**Reply by liodali** [reply]
yeah osmnavigation already using this project to get route between geopoints

**Reply by jesussmile** [reply]
I own a garmin watch, they use osm map too. They also have turn by turn navigation with time and distance available in offline mode. Its pretty accurate and fast for a watch with such low processing power. This is what got me interested and luckily stumbled upon your project. I wonder how they do it.  Not sure if it would help in anyway but thought I’d mention it.
